nm: add page

eddieantonio opened the pull request:
I did not use long options for compatibility between Apple and GNU `nm`. I limited examples to options that are available on both platforms.

igorshubovych commented on pages/common/nm.md:
```diff
@@ -0,0 +1,19 @@
+
+- Demangle C++ symbols (make them readable)
+
+`nm {{file.o}} | c++filt`
```
It is uncommon for us to have example of one command piped to another command.

eddieantonio replied:
Unfortunately, this is the only cross-platform way (between GNU, BSD) of doing C++ name demangling (GNU does have the [`--demangle` option](http://man7.org/linux/man-pages/man1/nm.1.html#OPTIONS)). Since `nm` output is less than helpful for C++ object files without name demangling (and, personally, the whole reason I'd use a tldr page for `nm` anyway), `c++filt` should be mentioned in here somewhere. Should it be put in a "see also" in the description?

waldyrious replied:
We do have a linux folder and are considering one for bsd (I suppose it would be ok if you created one yourself, /cc @rprieto), so the distinction could be done in that way instead of using the lowest common denominator in a common/ page.

waldyrious replied:
Or you could add a small note in the description with the workaround, although that would probably be frowned upon by some maintainers (cf. #321)

eddieantonio commented:
I created two pages: one for Linux and OS X. In the OS X version, I added a little "see c++filt"; for GNU/Linux, I added `--demangle`. I stress that for the tl;dr page, listing how to demangle C++ symbols is invaluable to have in nm's page.

igorshubovych commented:
LGTM. Thanks.
